compat/bsd_compat.h

```diff
@@ -176,14 +176,10 @@ char * strnstr(const char *s, const char *find, size_t slen);
 #endif
 
 #ifndef __unreachable
-# if defined(__GNUC__) && ((__GNUC__ >= 4) && (__GNUC_MINOR__ >= 6))
-#  define __unreachable() __builtin_unreachable()
-# elif defined(__clang__)
-#  if __has_builtin(__builtin_unreachable)
-#   define __unreachable() __builtin_unreachable()
-#  endif
+# if defined (__GNUC__) || defined (__clang__)
```

Comment:
Clang always defines __GNUC__.

IMHO we should just drop the if entirely. All modern gcc/clang support this construct. Modern here means all versions of the toolchain in all distros released in the last 5 years...

Reply:
I think it's fine to keep them. It's compile time check anyway.
Besides GCC/Clang, TinyCC, chibicc doesn't have these additional builtins, so compiling pkg with them will fail (although I haven't seen many people trying to do that).

Reply:
Tinyc also defines GNUC. And there's no chance any of these compilers could even come close to building. Trying to anticipate what's needed isn't going to help and history shows that they more often than not get in the way.  It wouldn't make tinyc work, for example, given the assumption gnuc being defines tells you anything useful. Tiny c is frustrating because it defines these things without being really compatible (but compatible enough for most tings)... without acually supporting those builds, it just adds needless complexity. They will still fail and you'll still have to make fixes to this actual code...